crates/csharp/src/function.rs

```diff
@@ -859,22 +862,37 @@ impl Bindgen for FunctionBindgen<'_, '_> {
                 );
                 let ret_area = self.locals.tmp("retArea");
 
-                self.needs_cleanup = true;
-                uwrite!(
-                    self.src,
-                    "
-                    void* {address};
-                    if (({size} * {list}.Count) < 1024) {{
-                        var {ret_area} = stackalloc {element_type}[({array_size}*{list}.Count)+1];
-                        {address} = (void*)(((int){ret_area}) + ({align} - 1) & -{align});
-                    }} 
-                    else
-                    {{
-                        var {buffer_size} = {size} * (nuint){list}.Count;
-                        {address} = NativeMemory.AlignedAlloc({buffer_size}, {align});
-                        cleanups.Add(()=> NativeMemory.AlignedFree({address}));
-                    }}
+                match realloc {
+                    None => {
+                        self.needs_cleanup = true;
+                        uwrite!(self.src,
+                            "
+                            void* {address};
+                            if (({size} * {list}.Count) < 1024) {{
+                                var {ret_area} = stackalloc {element_type}[({array_size}*{list}.Count)+1];
+                                {address} = (void*)(((int){ret_area}) + ({align} - 1) & -{align});
+                            }} 
+                            else
+                            {{
+                                var {buffer_size} = {size} * (nuint){list}.Count;
+                                {address} = NativeMemory.AlignedAlloc({buffer_size}, {align});
+                            }}
+                            cleanups.Add(()=> NativeMemory.AlignedFree({address}));
```

Comment:
This should be for the >= 1024 side of the if I think?

Reply:
And unit test for that use-case would be good.

Reply:
messed up on my rebase, good catch. I've also added a scenario to the integration tests that force the native alloc to happen.